code/modules/antagonists/cult/cult_structure_altar.dm

```diff
@@ -0,0 +1,36 @@
+/// Some defines for items the cult altar can create.
+#define ELDRITCH_WHETSTONE "Eldritch Whetstone"
+#define CONSTRUCT_SHELL "Construct Shell"
+#define UNHOLY_WATER "Flask of Unholy Water"
+
+// Cult altar. Gives out consumable items.
+/obj/structure/destructible/cult/item_dispenser/altar
+	name = "altar"
+	desc = "A bloodstained altar dedicated to Nar'Sie."
+	cult_examine_tip = "Can be used to create eldritch whetstones, construct shells, and flasks of unholy water."
+	icon_state = "talismanaltar"
+	break_message = "<span class='warning'>The altar shatters, leaving only the wailing of the damned!</span>"
+
+/obj/structure/destructible/cult/item_dispenser/altar/get_items_to_spawn(mob/living/user)
+	. = list()
+
+	var/list/items = list(
+		ELDRITCH_WHETSTONE = image(icon = 'icons/obj/kitchen.dmi', icon_state = "cult_sharpener"),
+		CONSTRUCT_SHELL = image(icon = 'icons/obj/wizard.dmi', icon_state = "construct_cult"),
+		UNHOLY_WATER = image(icon = 'icons/obj/drinks.dmi', icon_state = "holyflask")
+		)
+	var/choice = show_radial_menu(user, src, items, custom_check = CALLBACK(src, .proc/check_menu, user), require_near = TRUE, tooltips = TRUE)
+	switch(choice)
+		if(ELDRITCH_WHETSTONE)
+			. += /obj/item/sharpener/cult
+		if(CONSTRUCT_SHELL)
+			. += /obj/structure/constructshell
+		if(UNHOLY_WATER)
+			. += /obj/item/reagent_containers/glass/beaker/unholywater
```

Comment:
I feel like we could do this generically. List of define -> list(image, list(output) maybe?
Also you should make the radial options static

Reply:
for some reason i planned on making it static but never did
done now